### PR TITLE
SUBMARINE-1008. Submarine python code formatter and linter

### PR DESCRIPTION
**Pull request by KUAN-HSUN-LI**
### What is this PR for?
Currently, submarine python code format doesn't work well.
1. Replace `yapf` with `black`
2. Upgrade `isort`
3. Use `flake8` linter
4. Format the code in dev-support, submarine-sdk and website
5. Only check lint in dev-support, submarine-sdk since python code in website has its' style. (import not at top of file)
6. Update the SDK development guide

### What type of PR is it?
[Improvement]

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-1008

### How should this be tested?
I have updated the ci in GitHub Actions.

### Screenshots (if appropriate)
Original `dev-support/style-check/python/lint.sh` result. I have fixed all of the errors in the below images
螢幕擷取畫面 2021-09-06 094648
image

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? Yes
* Does this need new documentation? Yes

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -40,9 +40,9 @@ jobs:
           pip install --no-cache-dir torch==1.5.0
           pip install --no-cache-dir ./submarine-sdk/pysubmarine/.
           pip install -r ./submarine-sdk/pysubmarine/github-actions/test-requirements.txt
-          pip install -r ./submarine-sdk/pysubmarine/github-actions/lint-requirements.txt
+          pip install -r ./dev-support/style-check/python/lint-requirements.txt
       - name: Check python sdk code style
-        run: ./submarine-sdk/pysubmarine/github-actions/lint.sh
+        run: ./dev-support/style-check/python/lint.sh
```
Comment (pingsutw) [inline .github/workflows/python.yml:45]: Could we remove `lint-requirements.txt` and `lint.sh` in `./submarine-sdk/pysubmarine/github-actions/`

**Review by pingsutw — COMMENTED** [review]
Thank you for working on this. @KUAN-HSUN-LI 
Could you also update development docs?https://submarine.apache.org/docs/userDocs/submarine-sdk/pysubmarine/development

File: .github/workflows/python.yml (hunk 1 above)
Comment (KUAN-HSUN-LI) [reply .github/workflows/python.yml:45]: OK. I will remove it and also remove the `auto-format.sh` in `./submarine-sdk/pysubmarine/github-actions/`.

**Comment by KUAN-HSUN-LI** [comment]
@pingsutw Thanks for your review. I have removed the useless file and update the doc in `0.6.0` version.

**Comment by pingsutw** [comment]
Thanks~
Will merge it if no more comment

**Pull request closed without merging**